db/mysql/mysqlClient.go

```diff
@@ -0,0 +1,140 @@
+var ErrorFooFailedAdd = errors.New("failed to add")
+
+//ErrorFooFailedUpdate.
+var ErrorFooFailedUpdate = errors.New("failed to update")
+
+//ErrorFooFailedDelete.
+var ErrorFooFailedDelete = errors.New("failed to delete")
+
+//ErrorFooFailedRetrieve.
+var ErrorFooFailedRetrieve = errors.New("failed to retrieve")
+
+type mysqlStd struct {
+	db *sqlx.DB
+}
+
+func (m mysqlStd) AddStudent(student *models.Student) (string, error) {
+	if student.ID != "" {
+		return "", ErrorFooIDEmpty
+	}
+	student.ID = guuid.New().String()
+	names := student.Names()
+	if _, err := m.db.NamedExec(fmt.Sprintf(`INSERT INTO Student (%s) VALUES(%s)`, strings.Join(names, ","), strings.Join(mkPlaceHolder(names, ":", func(name, prefix string) string {
+		return prefix + name
+	}), ",")),
+		student); err != nil {
+		return "", ErrorFooFailedAdd
+	}
+	return "", nil
+}
+
+func (m mysqlStd) UpdateStudent(student *models.Student) error {
+	names := student.Names()
+	if _, err := m.db.NamedExec(fmt.Sprintf(`UPDATE Student SET %s`, strings.Join(mkPlaceHolder(names, "=:", func(name, prefix string) string {
+		return name + prefix + name
+	}), ",")), student); err != nil {
+		return ErrorFooFailedUpdate
+	}
+	return nil
+}
+
+func (m mysqlStd) GetStudentByID(id string) (*models.Student, error) {
+	var std models.Student
+	if err := m.db.Get(&std, fmt.Sprintf(`SELECT * FROM Student WHERE id='%s'`, id)); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, ErrorFooUserNotFoundError
+		}
+		return nil, err
+	}
+	fmt.Printf("%v\n", std.Map())
+	return &std, nil
+}
+
+func (m mysqlStd) ListStudent() ([]*models.Student, error) {
+	var stdList []*models.Student
+	var std models.Student
+	rows, err := m.db.Query(fmt.Sprintf(`SELECT * FROM Student`))
+	if err != nil {
+		if errors.Is(err, rows.Err()) {
+			return nil, ErrorFooFailedRetrieve
+		}
+	}
+	for rows.Next() {
+		err = rows.Scan(&std.ID, &std.Name, &std.Age, &std.Level, &std.Phone)
+		if err != nil {
+			return nil, ErrorFooFailedRetrieve
+		}
+		stdList = append(stdList, &std)
+	}
+	return stdList, nil
+}
+
+func (m mysqlStd) RemoveStudentByID(id string) error {
+	if _, err := m.db.Query(fmt.Sprintf(`DELETE FROM Student WHERE id= '%s'`, id)); err != nil {
+		return ErrorFooFailedDelete
+	}
+
+	return nil
+}
+
+func mkPlaceHolder(names []string, prefix string, formatName func(name, prefix string) string) []string {
+	ph := make([]string, len(names))
+	for i, name := range names {
+		ph[i] = formatName(name, prefix)
+	}
+	return ph
+}
+
+func formatDSN() string {
+	cfg := mysql.NewConfig()
+	cfg.Net = "tcp"
+	cfg.Addr = fmt.Sprintf("%s:%s", viper.GetString(config.DbHost), viper.GetString(config.DbPort))
+	cfg.DBName = viper.GetString(config.DbName)
+	cfg.ParseTime = true
+	cfg.User = viper.GetString(config.DbUser)
+	cfg.Passwd = viper.GetString(config.DbPass)
+	fmt.Printf("%v\n", cfg.FormatDSN())
+	return cfg.FormatDSN()
+}
+
+//NewMysqlfactory to connect DB
+func NewMysqlfactory(conf db.Option) (db.DataStore, error) {
```

Comment:
It's not really a factory.. it's just a client exporting function. Either use `NewMysqlClient` or `NewClient`.